### PR TITLE
Fix multiple comment delimiter break toggle comment shortcut

### LINKED ISSUE #83331: Code editor toggle comment doesn't uncomment if there's a different amount of #'s
### Godot version

v4.2.beta1.official [b1371806a]

### System information

Windows 10

### Issue description

Toggle comment adds more comment delimiters when there's different amount of delimiters, even if all the lines are commented.

Current behavior:
comment

Expected behavior (from v4.2.dev6.official [57a6813bb]):
comment_old

Regression likely happened in #81486

### Steps to reproduce

Open script editor. Comment some lines. Then select bigger area enclosing some of the commented lines, and toggle comment twice or more.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by Lunarisnia**
Fixes #83331

### CODE REVIEW

File: editor/code_editor.cpp [hunk 1]
```diff
@@ -1505,7 +1505,8 @@ void CodeTextEditor::toggle_inline_comment(const String &delimiter) {
 			// Empty lines should not be counted.
 			bool is_empty = text_editor->get_line(line).strip_edges().is_empty();
 			is_all_empty = is_all_empty && is_empty;
-			if (!is_empty && (delimiter_idx == -1 || text_editor->get_delimiter_start_key(delimiter_idx) != delimiter)) {
+			// `.substr(0, 1)` here because get_delimiter_start_key will return `##` instead of `#` when there is multiple comment delimiter in a line
+			if (!is_empty && (delimiter_idx == -1 || text_editor->get_delimiter_start_key(delimiter_idx).substr(0, 1) != delimiter)) {
```
Comment (AThousandShips) [inline editor/code_editor.cpp:1509]:
```suggestion
			if (!is_empty && (delimiter_idx == -1 || text_editor->get_delimiter_start_key(delimiter_idx).left(1) != delimiter)) {
```

**Review by SaracenOne — APPROVED** [review]
Tested, does what it says, LGTM

**Comment by akien-mga** [comment]
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by Lunarisnia** [comment]
sure thing!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**